### PR TITLE
Paygo aws pipeline

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR ?

Add a new deployment description using paygo instances
 - server paygo
 - proxy byos
 - sles15sp5 paygo
 - sles12sp5 paygo
 - slesforsap15sp5 paygo

This new deployment is needed to test paygo in AWS using testsuite.

 Remove known unsupported clients for paygo setup: 
 - sles15sp3 and sles15sp2 LTSS not available with paygo instances
  - ubuntu products not available by default
  - retail branch
  - no sle micro available

## Depends on : 
 - https://github.com/uyuni-project/sumaform/pull/1426
 - https://github.com/uyuni-project/sumaform/pull/1418
 - https://github.com/uyuni-project/sumaform/pull/1428
 - https://github.com/uyuni-project/sumaform/pull/1430
 - https://github.com/uyuni-project/sumaform/pull/1431
 - https://github.com/uyuni-project/sumaform/pull/1446
 - https://github.com/uyuni-project/sumaform/pull/1445
 - https://github.com/uyuni-project/sumaform/pull/1444
 - https://github.com/uyuni-project/sumaform/pull/1440
 - https://github.com/uyuni-project/sumaform/pull/1437
 - https://github.com/SUSE/spacewalk/pull/23121

### CODE REVIEW

**Review by ktsamis — APPROVED** [review]
I don't see any issues, looks good

**Review by Bischoff — APPROVED** [review]

File: jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy [hunk 1]
```diff
@@ -608,7 +633,11 @@ def getNodesHandler() {
     String[] moduleList = modules.split("\n")
     moduleList.each { lane ->
         def instanceList = lane.tokenize(".")
-        if (instanceList[1].contains('minion') || instanceList[1].contains('client')) {
```
Comment (srbarrios) [inline jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy:611]: What about using:
`def nodeName = lane.tokenize(".")[1]`

And then you can replace `instanceList[1]` by `nodeName` ?

File: terracumber_config/tf_files/SUSEManager-4.3-build-validation-paygo-AWS.tf [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+
+  sle12sp5_client_configuration    = module.sles12sp5-client.configuration
+  sle12sp5_minion_configuration    = module.sles12sp5-minion.configuration
+  sle12sp5_sshminion_configuration = module.sles12sp5-sshminion.configuration
+
+  sle15sp4_client_configuration    = module.sles15sp4-client.configuration
+  sle15sp4_minion_configuration    = module.sles15sp4-minion.configuration
+  sle15sp4_sshminion_configuration = module.sles15sp4-sshminion.configuration
+
+  sle15sp5_minion_configuration    = module.sles15sp5-minion.configuration
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-4.3-build-validation-paygo-AWS.tf:476]: Are we using the no-paygo minions anywhere for this PAYGO AWS Pipeline?
Would it make sense to test something not PAYG related on this pipeline?

File: jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy [hunk 3]
```diff
@@ -288,6 +287,29 @@ def run(params) {
...
+            /** PAYGO stages begin **/
+            if (params.enable_paygo_stages) {
+                // Call the minion testing.
+                try {
+                    stage('Clients paygo stages') {
+                        clientTestingStages(capybara_timeout, default_timeout, 'paygo')
+                    }
+
+                } catch (Exception ex) {
+                    println('ERROR: one or more clients have failed')
```
Comment (srbarrios) [inline jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy:299]:
```suggestion
                    println("ERROR: one or more clients have failed.\nException: ${ex}")
```

File: terracumber_config/tf_files/SUSEManager-4.3-build-validation-paygo-AWS.tf (hunk 2 above)
Comment (maximenoel8) [reply terracumber_config/tf_files/SUSEManager-4.3-build-validation-paygo-AWS.tf:476]: It doesn't cost much to add those minions and give us some feedback on paygo server. During my manual testing, I found some issues testing 'classic' minions with the paygo server. Just checking if the repositories are available is not enough in my opinion. Maybe we could limit it to minion. But it's really free to add those extra minions and I don't have server instability on AWS ( also with limited minions ).

**Pull request merged**